lib/pages/chat/events/video_player.dart

```diff
@@ -1,160 +1,208 @@
+    final height = widget.height ?? MessageContentStyle.imageHeight(context);
+
     return ClipRRect(
       borderRadius: BorderRadius.circular(12.0),
       child: Material(
         color: Colors.black,
         child: SizedBox(
-          width: MessageContentStyle.imageWidth(context),
-          height: MessageContentStyle.imageHeight(context),
-          child: chewieManager != null
-              ? FittedBox(
-                  fit: BoxFit.contain,
-                  child: Chewie(
-                    key: ValueKey(widget.event.eventId),
-                    controller: chewieManager,
+          width: MessageContentStyle.imageBubbleWidth(width),
+          height: MessageContentStyle.videoBubbleHeight(height),
+          child: Stack(
+            children: [
+              const BlurHash(hash: MessageContentStyle.defaultBlurHash),
+              if (hasThumbnail)
+                Center(
+                  child: ImageBubble(
+                    widget.event,
+                    tapToView: false,
+                    width: MessageContentStyle.imageBubbleWidth(width),
+                    height: MessageContentStyle.videoBubbleHeight(height),
                   ),
                 )
-              : Stack(
-                  children: [
-                    if (hasThumbnail)
-                      Center(
-                        child: ImageBubble(
-                          widget.event,
-                          tapToView: false,
-                        ),
-                      )
-                    else
-                      BlurHash(hash: blurHash),
-                    Center(
-                      child: OutlinedButton.icon(
-                        style: OutlinedButton.styleFrom(
-                          backgroundColor:
-                              Theme.of(context).colorScheme.surface,
-                        ),
-                        icon: _isDownloading
-                            ? const SizedBox(
-                                width: 24,
-                                height: 24,
-                                child: CircularProgressIndicator.adaptive(
-                                  strokeWidth: 2,
-                                ),
-                              )
-                            : const Icon(Icons.download_outlined),
-                        label: Text(
-                          _isDownloading
-                              ? L10n.of(context)!.loadingPleaseWait
-                              : L10n.of(context)!.videoWithSize(
-                                  widget.event.sizeString ?? '?MB',
-                                ),
-                        ),
-                        onPressed: _isDownloading ? null : _downloadAction,
-                      ),
-                    ),
-                  ],
+              else
+                BlurHash(hash: blurHash),
+              Center(
+                child: ValueListenableBuilder<DownloadState>(
```

Comment:
can use export this widget, input is DownloadState, so I can reuse it

Reply:
the widget `EventVideoPlayer` can be reused, so no need

Reply:
it's video_player class
<img width="774" alt="Screenshot 2023-09-22 at 14 33 04" src="https://github.com/linagora/twake-on-matrix/assets/43041967/002338e3-8b65-46b6-82be-7abdf9a4af15">

Reply:
so you should rename this file